rust_icu_ecma402/src/datetimeformat.rs

```diff
@@ -29,13 +29,179 @@ pub struct DateTimeFormat {
 
 pub(crate) mod internal {
     use ecma402_traits::datetimeformat::DateTimeFormatOptions;
+    use ecma402_traits::datetimeformat::options;
     use rust_icu_common as common;
     use rust_icu_ustring as ustring;
+    use rust_icu_uloc as uloc;
     use std::convert::TryFrom;
+    use anyhow::anyhow;
 
-    // TODO: implement this conversion completely.
-    pub fn opt_to_pattern(_opts: DateTimeFormatOptions) -> Result<ustring::UChar, common::Error> {
-        ustring::UChar::try_from("YYYY")
+    fn check_auto_styling_set(opts: &DateTimeFormatOptions) -> Result<(), common::Error> {
+        if opts.date_style.is_some() || opts.time_style.is_some() {
+            return Err(common::Error::Wrapper(anyhow!("date_style or time_style is set")));
```

Comment:
The error message isn't entirely clear. Maybe something like "can't use auto styling because date_style or time_style is set".